sdk/include/platform/sunburst/platform-spi.hh

```diff
@@ -0,0 +1,175 @@
+#pragma once
+#include <cdefs.h>
+#include <stdint.h>
+
+/**
+ * A Simple Driver for the Sonata's SPI.
+ *
+ * Documentation source can be found at:
+ * https://github.com/lowRISC/sonata-system/blob/1a59633d2515d4fe186a07d53e49ff95c18d9bbf/doc/ip/spi.md
+ *
+ * Rendered documentation is served from:
+ * https://lowrisc.org/sonata-system/doc/ip/spi.html
+ */
+struct SonataSpi
+{
+	/// Interrupt State Register
+	uint32_t interruptState;
+	/// Interrupt Enable Register
+	uint32_t interruptEnable;
+	/// Interrupt Test Register
+	uint32_t interruptTest;
+	/// Configuration register. Controls how the SPI block transmits and
+	/// receives data. This register can only be modified whilst the SPI block
+	/// is idle.
+	uint32_t configuration;
+	/// Controls the operation of the SPI block. This register can
+	/// only be modified whilst the SPI block is idle.
+	uint32_t control;
+	/// Status information about the SPI block
+	uint32_t status;
+	/// When written begins an SPI operation.
+	/// Writes are ignored when the SPI block is active.
+	uint32_t start;
+	/// Data from the receive FIFO. When read the data is popped from the FIFO.
+	/// If the FIFO is empty data read is undefined.
+	uint32_t receiveFifo;
+	/// Bytes written here are pushed to the transmit FIFO. If the FIFO is full
+	/// writes are ignored.
+	uint32_t transmitFifo;
+
+	/// Configuration Register Fields
+	enum : uint32_t
+	{
+		/// The length of a half period (i.e. positive edge to negative edge) of
+		/// the SPI clock, measured in system clock cycles reduced by 1. At the
+		/// standard Sonata 50 MHz system clock a value of 0 gives a 25 MHz SPI
+		/// clock, a value of 1 gives a 12.5 MHz SPI clock, a value of 2 gives
+		/// a 8.33 MHz SPI clock and so on.
+		ConfigurationHalfClkPeriodMask = 0xffu << 0,
+		/// When set the most significant bit (MSB) is the first bit sent and
+		/// received with each byte
+		ConfigurationMSBFirst = 1u << 29,
+		/// The phase of the spi_clk signal. When CPHA is 0 data is sampled on
+		/// the leading edge and changes on the trailing edge. The first data
+		/// bit is immediately available before the first leading edge of the
+		/// clock when transmission begins. When CPHA is 1 data is sampled on
+		/// the trailing edge and change on the leading edge.
+		ConfigurationCPHA = 1u << 30,
+		/// The polarity of the spi_clk signal. When CPOL is 0 clock is low when
+		/// idle and the leading edge is positive. When CPOL is 1 clock is high
+		/// when idle and the leading edge is negative
+		ConfigurationCPOL = 1u << 31,
+	};
+
+	/// Control Register Fields
+	enum : uint32_t
+	{
+		/// Write 1 to clear the transmit FIFO.
+		ControlTxClear = 1 << 0,
```

Comment:
```suggestion
		ControlTransmitClear = 1 << 0,
```

The abbreviations tx and rx are ubiquitous in communication literature, but not everyone reading this code will have that background.

Same comment for the next ones down.